pkg/actuators/machine/reconciler.go

```diff
@@ -39,6 +39,20 @@ func newReconciler(scope *machineScope) *Reconciler {
 func (r *Reconciler) create() error {
 	klog.Infof("%s: creating machine", r.machine.Name)
 
+	if instances, err := r.getMachineInstances(); err == nil && len(instances) > 0 {
```

Comment:
Here we also trying to find vms based on tags. I'm bit afraid about low probable but undesirable consequences.
Might it make sense to swap this with block below and check InstanceID presence first?

Reply:
If we get an error, then we fall back to the `instanceID` check on line 49 which will prevent us from creating a secondary machine and we will requeue. Then `Exists` on the next reconcile will either succeed or get the same 500, and it will be handled

> Here we also trying to find vms based on tags. I'm bit afraid about low probable but undesirable consequences.
Might it make sense to swap this with block below and check InstanceID presence first?

If we find the instance based on tags, that's fine, we found the instance, if we find the instance, we requeue, Exists passes next time and then we go into the Update cycle

Reply:
ok, thx for explanation